### ISSUE TITLE
Firefox, RStudio Server, and latest Shiny/httpuv don't work together

### ISSUE THREAD

**Issue by jcheng5** [rstudio/rstudio#2940]
TL;DR: [This line](https://github.com/rstudio/rstudio/blob/415a39c1c7e7c03787d2b8d18fa36c73cd05dc53/src/cpp/server/ServerSessionProxy.cpp#L805) doesn't account for the fact that some browsers send `Connection: Upgrade` and others (Firefox) send `Connection: keep-alive, Upgrade`.

---

With the combination of httpuv 1.4.x, RStudio Server, and Firefox, running Shiny apps through the session proxy doesn't work. The problem is fixed if you change ANY of those three though (switching to older httpuv, OR RStudio Desktop, OR Chrome all fix it).

The problem is with the HTTP request that initiates a WebSocket connection, and specifically, with that request's Connection header. Chrome sends `Connection: Upgrade`, while Firefox sends `Connection: keep-alive, Upgrade`. RFC 6455 page 17 says:

> The request MUST contain a |Connection| header field whose value MUST include the "Upgrade" token.

so both browsers are passing legal values.

However, RStudio Server's [proxy logic assumes](https://github.com/rstudio/rstudio/blob/415a39c1c7e7c03787d2b8d18fa36c73cd05dc53/src/cpp/server/ServerSessionProxy.cpp#L805) that the incoming request's Connection header must be equal to "Upgrade" to be considered an upgrade, and if it is not, it changes the header to `Connection: close` before sending upstream to httpuv.

httpuv <1.4 doesn't care if `Connection: close`, it allows a websocket connection anyway. That's why it can work with either Firefox or Chrome. But httpuv 1.4.x looks for `Connection: Upgrade`, so it doesn't work with Firefox.

### System details

    RStudio Edition : Server
    RStudio Version : 1.2.704 and 1.1.453
    OS Version      : Ubuntu 16.04
    R Version       : 3.4? Doesn't matter

### Steps to reproduce the problem

1. Use Firefox (must be Firefox) to log into RStudio Server
1. Install Shiny from CRAN (httpuv 1.4.0 or higher must be used)
2. Start a Shiny app, e.g. `shiny::runExample("01_hello")`, and let new browser window launch

### Describe the problem in detail

The app's page goes grey immediately

### Describe the behavior you expected

The app should work correctly

(cc @wch)
